File: Practical10/app/src/main/java/com/example/practical10/NoteDao.java
package com.example.practical10;

import android.provider.ContactsContract;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface NoteDao {
    @Insert
    void insert(Note note);

    @Delete
    void delete(Note note);

    @Query("SELECT * FROM notes")
    List<Note> getAllNotes();


    // use : notation to represent parsed data
    @Query("SELECT * FROM notes WHERE notes.id = :id")
    Note getNoteById(int id);

}

File: KotlinPractical12/app/src/main/java/com/example/kotlinpractical12/ui/data/NoteDao.kt
package com.example.kotlinpractical12.ui.data

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface NoteDao {

    @Insert (onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(note: Note)

    @Delete
    suspend fun delete(note: Note)

    @Query("SELECT * FROM note WHERE id == :noteid")
    fun getNoteById(noteid: Int): Flow<Note>

    @Query("SELECT * FROM note")
    fun getAll(): Flow<List<Note>>

}